plugins/woo-ai/api/text-generation/class-text-generation.php

```diff
@@ -0,0 +1,172 @@
+		$this->register_routes();
+	}
+
+	/**
+	 * Endpoint namespace.
+	 *
+	 * @var string
+	 */
+	protected $namespace = 'wooai';
+
+	/**
+	 * Route base.
+	 *
+	 * @var string
+	 */
+	protected $rest_base = 'text-generation';
+
+	/**
+	 * Register routes.
+	 */
+	public function register_routes() {
+		register_rest_route(
+			$this->namespace,
+			'/' . $this->rest_base,
+			array(
+				array(
+					'methods'             => \WP_REST_Server::CREATABLE,
+					'callback'            => array( $this, 'get_response' ),
+					'permission_callback' => array( $this, 'get_response_permission_check' ),
+					'args'                => array(
+						'prompt'            => array(
+							'type'              => 'string',
+							'validate_callback' => 'rest_validate_request_arg',
+							'required'          => true,
+						),
+						'temperature'       => array(
+							'type'              => 'number',
+							'validate_callback' => 'rest_validate_request_arg',
+						),
+						'previous_messages' => array(
+							'description' => __( 'Optional parameter to get only specific task lists by id.', 'woocommerce' ),
+							'type'        => 'array',
+						),
+					),
+				),
+			)
+		);
+	}
+
+	/**
+	 * Check if a given request has access to read.
+	 *
+	 * @param  WP_REST_Request $request Full details about the request.
+	 * @return WP_Error|boolean
+	 */
+	public function get_response_permission_check( $request ) {
+		if ( ! wc_rest_check_post_permissions( 'product', 'create' ) ) {
+			return new \WP_Error( 'woocommerce_rest_cannot_create', __( 'Sorry, you are not allowed to create resources.', 'woocommerce' ), array( 'status' => rest_authorization_required_code() ) );
+		}
+
+		return true;
+	}
+
+	/**
+	 * Get the onboarding tasks.
+	 *
+	 * @param WP_REST_Request $request Full details about the request.
+	 * @return WP_REST_Response|WP_Error
+	 */
+	public function get_response( $request ) {
+		$prompt            = $request->get_param( 'prompt' );
+		$temperature       = $request->get_param( 'temperature' ) ? $request->get_param( 'temperature' ) : 1;
+		$previous_messages = $request->get_param( 'previous_messages' ) ? $request->get_param( 'previous_messages' ) : array();
+		$request_model     = $request->get_param( 'model' ) ? $request->get_param( 'model' ) : 'openai';
+
+		$model = $this->models[ $request_model ];
+
+		$api_url = $model['url'];
+		$api_key = $model['key'];
+
+		$messages   = $previous_messages;
+		$messages[] = array(
+			'role'    => 'user',
+			'content' => $prompt,
+		);
+
+		$post_data = array(
+			'messages'    => $messages,
+			'model'       => 'gpt-3.5-turbo',
+			'temperature' => $temperature,
+		);
+
+		$raw_response = wp_remote_post(
+			$api_url,
+			array(
+				'method'  => 'POST',
+				'body'    => wp_json_encode( $post_data ),
+				'timeout' => 45,
+				'headers' => array(
+					'Content-Type'  => 'application/json',
+					'Authorization' => 'Bearer ' . $api_key,
+				),
+			)
+		);
+
+		if ( is_wp_error( $raw_response ) ) {
```

Comment:
Good call. I've added a simple version of this in https://github.com/woocommerce/woocommerce/pull/38309/commits/d9ca739dda56878a07d9f164d057b7f6d368eb03 , although this might be influenced by the WPCOM APIs when we switch over to those.